### PR TITLE
Feature/df 831  job favourite item es document

### PR DESCRIPTION
**Pull request by pavax**

### CODE REVIEW

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/favouriteitem/FavouriteItemAuthorizationService.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+@Component
+@Transactional
+public class FavouriteItemAuthorizationService {
+
+    private final CurrentUserContext currentUserContext;
+
+    private final FavouriteItemRepository favouriteItemRepository;
+
+    private final JobAdvertisementRepository jobAdvertisementRepository;
```
Comment (birom) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/favouriteitem/FavouriteItemAuthorizationService.java]: Please remove unused variable.

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/common/ElasticsearchIndexService.java [hunk 2]
```diff
@@ -50,38 +48,96 @@
...
+                    @Override
+                    public Iterable<ApiUserDocument> saveAll(Iterable<ApiUserDocument> entities) {
+                        return apiUserElasticsearchRepository.saveAll(entities);
+                    }
+                },
+                ApiUserDocument::new,
+                "streamAll"
+        );
+
+        reindexForClass(
```
Comment (birom) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/common/ElasticsearchIndexService.java]: This call will drop the ``job-advertisements`` index again after reindexing the ``JobAdvertisementDocument``.

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/favouriteitem/FavouriteItemAlreadyExists.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package ch.admin.seco.jobs.services.jobadservice.application.favouriteitem;
+
+import ch.admin.seco.jobs.services.jobadservice.domain.favouriteitem.FavouriteItemId;
+import ch.admin.seco.jobs.services.jobadservice.domain.jobadvertisement.JobAdvertisementId;
+
+class FavouriteItemAlreadyExists extends RuntimeException {
+
+    FavouriteItemAlreadyExists(FavouriteItemId favouriteItemId, JobAdvertisementId jobAdvertisementId, String ownerid) {
+        super("FavouriteItem couldn't be created. User: '" + ownerid + "' already has a FavouriteItem: '" + favouriteItemId.getValue() + "' for JobAdvertismentId: '" + jobAdvertisementId.getValue() + "'");
```
Comment (cjpsycos) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/favouriteitem/FavouriteItemAlreadyExists.java]: userId or ownerUserId

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/favouriteitem/FavouriteItemAuthorizationService.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    private final CurrentUserContext currentUserContext;
+
+    private final FavouriteItemRepository favouriteItemRepository;
+
+    public FavouriteItemAuthorizationService(CurrentUserContext currentUserContext, FavouriteItemRepository favouriteItemRepository) {
+        this.currentUserContext = currentUserContext;
+        this.favouriteItemRepository = favouriteItemRepository;
+    }
+
+    public boolean matchesCurrentUserId(String ownerId) {
```
Comment (cjpsycos) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/favouriteitem/FavouriteItemAuthorizationService.java]: ownerUserId

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/favouriteitem/FavouriteItem.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    private LocalDateTime createdTime;
+
+    @NotNull
+    private LocalDateTime updatedTime;
+
+    @Size(max = 1000)
+    private String note;
+
+    @NotBlank
+    private String ownerId;
```
Comment (cjpsycos) [inline domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/favouriteitem/FavouriteItem.java]: ownerUserId

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/favouriteitem/FavouriteItemAlreadyExists.java (hunk 3 above)
Comment (pavax) [reply application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/favouriteitem/FavouriteItemAlreadyExists.java]: userId

**Pull request merged**